### PR TITLE
enhance: add NewSegmentWithLoadInfo API to support segment self-managed loading

### PR DESCRIPTION
**Pull request by congqixia**
This commit introduces the foundation for enabling segments to manage their own loading process by passing load information during segment creation.

Changes:

C++ Layer:
- Add NewSegmentWithLoadInfo() C API to create segments with serialized load info
- Add SetLoadInfo() method to SegmentInterface for storing load information
- Refactor segment creation logic into shared CreateSegment() helper function
- Add comprehensive documentation for the new API

Go Layer:
- Extend CreateCSegmentRequest to support optional LoadInfo field
- Update segment creation in querynode to pass SegmentLoadInfo when available
- Add ConvertToSegcoreSegmentLoadInfo() and helper converters for proto translation

Proto Definitions:
- Add segcorepb.SegmentLoadInfo message with essential loading metadata
- Add supporting messages: Binlog, FieldBinlog, FieldIndexInfo, TextIndexStats, JsonKeyStats
- Remove dependency on data_coord.proto by creating segcore-specific definitions

Testing:
- Add comprehensive unit tests for proto conversion functions
- Test edge cases including nil inputs, empty data, and nil array/map elements

This is the first step toward issue #45060 - enabling segments to autonomously manage their loading process, which will:
- Clarify responsibilities between Go and C++ layers
- Reduce cross-language call overhead
- Enable precise resource management at the C++ level
- Support better integration with caching layer
- Enable proactive schema evolution handling

Related to #45060

### CODE REVIEW

**Comment by congqixia** [comment]
rerun go-sdk

**Comment by congqixia** [comment]
rerun go-sdk

File: internal/core/src/segcore/segment_c.cpp [hunk 1]
```diff
@@ -90,6 +111,34 @@ NewSegment(CCollection collection,
...
+        AssertInfo(load_info_blob, "load info is null");
+        milvus::proto::segcore::SegmentLoadInfo load_info;
+        auto suc = load_info.ParseFromArray(load_info_blob, load_info_length);
+        AssertInfo(suc, "unmarshal load info failed");
+
+        auto col = static_cast<milvus::segcore::Collection*>(collection);
+
+        auto segment =
+            CreateSegment(col, seg_type, segment_id, is_sorted_by_pk);
+        // segment->LoadFieldData(load_info);
```
Comment (zhengbuqian) [inline internal/core/src/segcore/segment_c.cpp]: should we uncomment this line?

File: pkg/proto/segcore.proto [hunk 2]
```diff
@@ -46,3 +81,43 @@ message CollectionIndexMeta {
...
+  repeated string index_file_paths = 7;
+  int64 index_size = 8;
+  int64 index_version = 9;
+  int64 num_rows = 10;
+  int32 current_index_version = 11;
+  int64 index_store_version = 12;
+}
+
+message SegmentLoadInfo {
+  reserved 14, 15, 17;  // Removed: start_position, delta_position, level
```
Comment (zhengbuqian) [inline pkg/proto/segcore.proto]: is it necessary to keep the structure the same as querypb counterparts? we will not try to directly parse a segcorepb.SegmentLoadInfo from querypb.SegmentLoadInfo bytes?

File: pkg/proto/segcore.proto [hunk 3]
```diff
@@ -5,6 +5,41 @@ option go_package = "github.com/milvus-io/milvus/pkg/v2/proto/segcorepb";
 import "schema.proto";
 import "common.proto";
 
+message Binlog {
+  int64 entries_num = 1;
+  uint64 timestamp_from = 2;
+  uint64 timestamp_to = 3;
+  string log_path = 4 [deprecated = true];
```
Comment (zhengbuqian) [inline pkg/proto/segcore.proto]: ditto: if no longer needed, we can remove this in the new proto.

File: pkg/proto/segcore.proto (hunk 2 above)
Comment (congqixia) [reply pkg/proto/segcore.proto]: just try to keep this possibility if convertion consume too much time

**Comment by congqixia** [comment]
rerun go-sdk

**Review by zhengbuqian — COMMENTED** [review]
/lgtm
